core/src/main/resources/jenkins/model/BuildDiscarderProperty/help.html

```diff
@@ -1,19 +1,51 @@
 <div>
-  This controls the disk consumption of Jenkins by managing how long you'd like to keep
-  records of the builds (such as console output, build artifacts, and so on.)
-  Jenkins offers two criteria:
-
+  This determines when, if ever, build records for this project should be
+  discarded. Build records include the console output, archived artifacts, and
+  any other metadata related to a particular build.
+  <p>
+  Keeping fewer builds means less disk space will be used in the <i>Build Record
+  Root Directory</i>, which is specified on the <i>Configure System</i> screen.
+  <p>
+  <!-- TODO: Most/all of the following should be dynamic, showing info from the
+       `BuildDiscarder` implementations that are currently loaded -->
+  Jenkins offers two options for determining when builds should be discarded:
```

Comment:
This is an extension point, needs to be mentioned. Currently it smells like 2 hardcoded strategies